daemon/cmd/daemon_main.go

```diff
@@ -1768,6 +1771,16 @@ func runDaemon() {
 			})
 			ms.CollectStaleMapGarbage()
 			ms.RemoveDisabledMaps()
+
+			if len(d.restoredCIDRs) > 0 {
+				// Release restored CIDR identities after a grace period (default 10
+				// minutes).  Any identities actually in use will still exist after
+				// this.
+				time.Sleep(option.Config.IdentityRestoreGracePeriod)
+				log.Debugf("Releasing reference counts for %d restored CIDR identities", len(d.restoredCIDRs))
+
+				d.ipcache.ReleaseCIDRIdentitiesByCIDR(d.restoredCIDRs)
```

Comment:
I forgot on my initial review. `d.restoredCIDRs` is a slice which we don't know how big it is but I am assuming it can be big in certain clusters. After we are done with `d.ipcache.ReleaseCIDRIdentitiesByCIDR(d.restoredCIDRs)` can we can free up the memory usage by running `d.restoredCIDRs = nil`?

Reply:
Done.

Reply:
I made the sleep conditional on `!k8s.IsEnabled()` and adjusted the help text on the new CLI option accordingly.

Reply:
Like this:
```
@@ -1776,10 +1776,20 @@ func runDaemon() {
 				// Release restored CIDR identities after a grace period (default 10
 				// minutes).  Any identities actually in use will still exist after
 				// this.
-				time.Sleep(option.Config.IdentityRestoreGracePeriod)
+				//
+				// This grace period is not needed when running with K8s, as in that
+				// case we get here only after all state has been sychronized from
+				// K8s. If running on an external workload, or in any other case
+				// where k8s is not configured we need a grace period during which
+				// policies can be restored by importing them usign the Cilium API.
+				if !k8s.IsEnabled() {
+					time.Sleep(option.Config.IdentityRestoreGracePeriod)
+				}
 				log.Debugf("Releasing reference counts for %d restored CIDR identities", len(d.restoredCIDRs))
 
 				d.ipcache.ReleaseCIDRIdentitiesByCIDR(d.restoredCIDRs)
+				// release the memory held by restored CIDRs
+				d.restoredCIDRs = nil
 			}
```